alacritty/src/daemon.rs

```diff
@@ -58,18 +59,21 @@ where
 {
     let mut command = Command::new(program);
     command.args(args).stdin(Stdio::null()).stdout(Stdio::null()).stderr(Stdio::null());
-    if let Ok(cwd) = foreground_process_path(master_fd, shell_pid) {
-        command.current_dir(cwd);
-    }
+
     unsafe {
         command
-            .pre_exec(|| {
+            .pre_exec(move || {
                 match libc::fork() {
                     -1 => return Err(io::Error::last_os_error()),
                     0 => (),
                     _ => libc::_exit(0),
                 }
 
+                // Don't fail spawning when we can't change the working directory.
+                if let Ok(cwd) = foreground_process_path(master_fd, shell_pid) {
```

Comment:
While `foreground_process_path` should just call `readlink`, which is async-signal-safe, I think it's super easy to accidentally break things here.

Since there's no need to call this function inside `pre_exec`, could you move it outside and just assign it to a temporary variable?